api/src/workers/base-event-handler.ts

```diff
@@ -55,38 +53,45 @@ abstract class BaseEventHandler {
       // when the gap is too narrow, it returns an error.
       const blockDiff = toBlock - fromBlock;
       if (blockDiff < this.blockThreshold) {
-        console.log("skipping block, blockDiff = ", blockDiff);
+        console.log("Skipping block, blockDiff = ", blockDiff);
         continue;
       }
 
       // do our processing
       let getEventsResult, eventList;
 
       try {
-        // `getEvents` will retrieve all events of the given type within the block height range supplied.
+        // `getEventsResult` will retrieve all events of the given type within the block height range supplied.
         // See https://docs.onflow.org/core-contracts/access-api/#geteventsforheightrange
-        getEventsResult = await send([
-          getEvents(this.eventName, fromBlock, toBlock),
-        ]);
-        eventList = await fcl.decode(getEventsResult);
+        getEventsResult = await Promise.all(
+          this.eventNames.map((name) =>
+            send([getEvents(name, fromBlock, toBlock)])
+          )
+        ).catch((e) => {
+          console.log("Error getting events:", e);
+        });
+
+        eventList = await Promise.all(
+          getEventsResult.map((result) => fcl.decode(result))
+        ).catch((e) => {
+          console.log("Error creating event list:", e);
+        });
+
+        eventList = eventList.flat();
       } catch (e) {
         console.error(
-          `error retrieving events for block range fromBlock=${fromBlock} toBlock=${toBlock}`,
+          `Error retrieving events for block range fromBlock=${fromBlock} toBlock=${toBlock}`,
           e
         );
         continue;
       }
 
-      for (let i = 0; i < eventList.length; i++) {
-        console.log(
-          "block_height =",
-          getEventsResult.events[i].blockHeight,
-          " payload =",
-          eventList[i].data
-        );
-        const blockHeight = getEventsResult.events[i].blockHeight;
-        await this.processAndCatchEvent(blockHeight, eventList, i);
-      }
+      // Note: Events may be "processed" out of order.
```

Comment:
Maybe this should be done sequentially (without `Promise.all`)? As of now there are no logical dependencies between events, but based on my experience working with Ethereum events for CK, I think that could very likely change in the future. 

I also could see many extensions of this example needing to ensure strict ordering when processing events, and they might not realize that this handler sends them through in parallel.

Reply:
Yes, i had this on a TODO as well. We can make this sequential.